### PR TITLE
build forms with Ruby on rails

### PR DESCRIPTION
**Pull request by ashraffares**
using rails form helper 
form _tag
form_with

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  resources :users#, only: [:new, :create, :edit, :update,:show]
```
Comment (voscarmv) [inline config/routes.rb]: Please use the routes specified by The Odin Project here: 
```
 resources :users, only: [:new, :create, :edit, :update]
```
image

image
https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work. 💻 👀 ✨ 

## However, there are some changes required 👀 🧰 🔨 🔧 
- Please add some validation to your `edit` interface. Currently, I'm able to change the username to a blank username. This should not be possible. Please fix it.
- Please check my code comments for more.

**[OPTIONAL]**
- You may create a Rails `partial` for the user form which is reused in the `new` and `edit` views. 
- You may add proper error reporting so that the user knows what went wrong when filling out the form in case it does not pass validation.
- You may also validate the e-mail address format with a regular expression or the built-in URI method.
### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

File: config/routes.rb (hunk 1 above)
Comment (ashraffares) [reply config/routes.rb]: thank a lot for your review I'm done with the required changes 
- change routes file     resources :users, only: [:new, :create, :edit, :update]
- Add validation for email
- Display alert messages

**Review by oloomoses — APPROVED** [review]
# Congratulations, Milestone Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**


**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Pull request merged**